waku/v2/peermanager/peer_manager.go

```diff
@@ -420,7 +420,7 @@ func (pm *PeerManager) addPeer(ID peer.ID, addrs []ma.Multiaddr, origin wps.Orig
 }
 
 // AddPeer adds peer to the peerStore and also to service slots
-func (pm *PeerManager) AddPeer(address ma.Multiaddr, origin wps.Origin, pubSubTopics []string, protocols ...protocol.ID) (peer.ID, error) {
+func (pm *PeerManager) AddPeer(address ma.Multiaddr, origin wps.Origin, pubSubTopics []string, connectNow bool, protocols ...protocol.ID) (peer.ID, error) {
```

Comment:
To avoid adding an extra parameter to the function to connect (which is only used in a single place), what do you think if we refactor it to return the `PeerData` and also split it into two separate functions, and modify AddrInfoToPeerData so it looks like this:


```go

func AddrInfoToPeerData(origin wps.Origin, peerID peer.ID, host host.Host, pubsubTopics ...string ) *service.PeerData {
	addrs := host.Peerstore().Addrs(peerID)
	if len(addrs) == 0 {
		//Addresses expired, remove peer from peerStore
		host.Peerstore().RemovePeer(peerID)
		return nil
	}
	return &service.PeerData{
		Origin: origin,
		AddrInfo: peer.AddrInfo{
			ID:    peerID,
			Addrs: addrs,
		},
               PubsubTopics: pubsubTopics,
	}
}

func (pm *PeerManager) AddPeer(address ma.Multiaddr, origin wps.Origin, pubSubTopics []string, protocols ...protocol.ID) (service.PeerData, error) {
	//Assuming all addresses have peerId
	info, err := peer.AddrInfoFromP2pAddr(address)
	if err != nil {
		return "", err
	}
	//Add Service peers to serviceSlots.
	for _, proto := range protocols {
		pm.addPeerToServiceSlot(proto, info.ID)
	}
	//Add to the peer-store
	err = pm.addPeer(info.ID, info.Addrs, origin, pubSubTopics, protocols...)
	if err != nil {
		return "", err
	}

        return AddrInfoToPeerData(origin, info.ID, pm.host, pubsubTopics...)
        return info, nil
}

func (pm *PeerManager) Connect(peerData service.PeerData) {
    go pm.peerConnector.PushToChan(peerData)
}
```
 

Then in waku_store_client.go, line 284, you'd do something like this:
```go
		peerData, err := store.pm.AddPeer(params.peerAddr, peerstore.Static, pubsubTopics, true, StoreID_v20beta4)
			if err != nil {
				return nil, err
			}

                        store.pm.Connect(peerData)

			params.selectedPeer = peerData.AddrInfo.ID
```

Reply:
Good idea, i remember wanting to add a connect method but for some reason backed out if it and did this.